home/private_dot_local/lib/agent-equipment/agent_equipment/authorization.py

```
@@ -0,0 +1,674 @@
+        )
+
+
+class FileAuthorizationLedger:
+    """One trusted execution-domain ledger backed by exclusive durable files."""
+
+    def __init__(
+        self,
+        root: Path,
+    ) -> None:
+        if not isinstance(root, Path) or not root.is_absolute() or ".." in root.parts:
+            raise ValueError("authorization ledger root must be one absolute path")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)